src/audio/component.c

```diff
@@ -326,121 +326,210 @@ int comp_verify_params(struct comp_dev *dev, uint32_t flag,
 	return 0;
 }
 
-void comp_free_model_data(struct comp_dev *dev, struct comp_model_data *model)
+/** \brief Struct handler for large component configs */
+struct comp_data_blob_handler {
+	uint32_t data_size;	/**< size of component's model data */
+	void *data;		/**< pointer to model data */
+	void *data_new;		/**< pointer to model data */
+	bool data_ready;	/**< set when fully received */
+	uint32_t data_pos;	/**< indicates a data position in data
+				  *  sending/receiving process
+				  */
+};
+
+static void comp_free_data_blob(struct comp_dev *dev,
+				struct comp_data_blob_handler *model)
 {
-	if (!model->data)
+	if (!model || !model->data)
 		return;
 
 	rfree(model->data);
+	rfree(model->data_new);
 	model->data = NULL;
+	model->data_new = NULL;
 	model->data_size = 0;
-	model->crc = 0;
-	model->data_pos = 0;
 }
 
-int  comp_alloc_model_data(struct comp_dev *dev, struct comp_model_data *model,
-			   uint32_t size)
+struct comp_data_blob comp_get_data_blob(struct comp_dev *dev,
+					 struct comp_data_blob_handler
+						*model_handler)
 {
-	comp_free_model_data(dev, model);
+	struct comp_data_blob config = {NULL, 0};
+
+	comp_dbg(dev, "comp_get_data_blob()");
+
+	if (!model_handler) {
+		comp_err(dev, "comp_get_data_blob(): !model_handler");
+		return config;
+	}
+
+	if (comp_is_new_data_blob_available(dev, model_handler)) {
```

Comment:
Please comment what this block does - along with **all** other blocsk too.

Reply:
@lgirdwood 
I've added a few comments. Please take a look whether it satisfies you.